src/Infrastructure.VS/ErrorListHelper.cs

```diff
@@ -40,78 +39,105 @@ public ErrorListHelper(IVsUIServiceOperation vSServiceOperation)
         {
             this.vSServiceOperation = vSServiceOperation;
         }
-
+        
         public bool TryGetRuleIdFromSelectedRow(out SonarCompositeRuleId ruleId)
         {
             SonarCompositeRuleId ruleIdOut = null;
-            var result = vSServiceOperation.Execute<SVsErrorList, IErrorList, bool>(
-                errorList =>
-                {
-                    return DoTryGetRuleIdFromSelectedRow(errorList, out ruleIdOut);
-                });
+            var result = vSServiceOperation.Execute<SVsErrorList, IErrorList, bool>(errorList =>
+                TryGetSelectedTableEntry(errorList, out var handle) && TryGetRuleId(handle, out ruleIdOut));
 
             ruleId = ruleIdOut;
 
             return result;
         }
-
-        private bool DoTryGetRuleIdFromSelectedRow(IErrorList errorList, out SonarCompositeRuleId ruleId)
+        
+        public bool TryGetRuleId(ITableEntryHandle handle, out SonarCompositeRuleId ruleId)
         {
-            ruleId = null;
-            var selectedItems = errorList?.TableControl?.SelectedEntries;
-
-            if (selectedItems?.Count() == 1)
-            {
-                var handle = selectedItems.First();
-                TryGetRuleId(handle, out ruleId);
-            }
-
-            return ruleId != null;
+            var errorCode = FindErrorCodeForEntry(handle);
+            return SonarCompositeRuleId.TryParse(errorCode, out ruleId);
         }
 
-        public bool TryGetRuleId(ITableEntryHandle tableEntryHandle, out SonarCompositeRuleId ruleId)
+        public bool TryGetIssueFromSelectedRow(out IFilterableIssue issue)
```

Comment:
Here's the main change of the PR. The other changes in this file are refactoring of the existing code to reuse it here